pinc/Quiz.inc

```diff
@@ -168,28 +168,30 @@ class Quiz
             {
                 $passed = user_has_passed_quiz_page($username, $quiz_page_id);
                 $text = $passed ? _("Passed") : _("Not passed");
-                $bgcolor = $passed ? '#ccffcc' : '#ffcccc';
-                echo "<td style='background-color: $bgcolor'>$text</td>";
+                $bgcolor = $passed ? 'pass' : 'no-pass';
+                echo "<td class='$bgcolor'>$text</td>";
+
                 $date = get_last_attempt_date_for_quiz_page($username, $quiz_page_id);
                 $text = ($date != 0) ? strftime("%d-%b-%y", $date) : _("Not attempted");
                 $max = $this->pass_requirements['maximum_age'];
                 $date_ok = ((time() - $date) < $max) || empty($max);
-                $bgcolor = $date_ok ? '#ccffcc' : '#ffcccc';
-                echo "<td style='background-color: $bgcolor'>$text</td>";
-                $text = ($passed && $date_ok) ?	"<span style='font-size: 1.5em' title='Quiz page passed!'>&check;</span>"
-                    : "<span style='font-size: 1.5em' title='Quiz page not passed'>&#x2717;</span>";
-                $bgcolor = ($passed && $date_ok) ? '#88ff88' : '#ff8888';
-                echo "<td style='background-color: $bgcolor'>$text</td>";
+                $bgcolor = $date_ok ? 'date-ok' : 'date-not-ok';
+                echo "<td class='$bgcolor'>$text</td>";
+
+                $text = ($passed && $date_ok) ?	"<span class='x-large' title='Quiz page passed!'>&check;</span>"
```

Comment:
I realize this was like this already, but it's a bit unwieldy as a ternary (not to mention the title isn't translated -- wonder if that was my fault in an earlier commit!).

Perhaps:
```php
if($passed && $date_ok)
{
    $text = "<span class='x-large' title='" . attr_safe(_("Quiz page passed!")) . "'>&check;</span>";
    $class = 'quiz-ok';
}
else
{
    $text = "<span class='x-large' title='" . attr_safe(_("Quiz page not passed!")) . "'>&#x2717;</span>";
    $class = 'quiz-not-ok';
}
```

Reply:
agree, if we go this way though, can we put unicode characters directly instead of their encodings? so '✗' and '✓'.

Reply:
> agree, if we go this way though, can we put unicode characters directly instead of their encodings? so '✗' and '✓'.

I have no foundation knowledge to know which is the better way to go -- however, after a bit of research, I see that the entities are used elsewhere in the code for WC purposes (pre-Unicode?).

Reply:
> I realize this was like this already, but it's a bit unwieldy as a ternary (not to mention the title isn't translated -- wonder if that was my fault in an earlier commit!).
> 
> Perhaps:
> 
> ```
> if($passed && $date_ok)
> {
>     $text = "<span class='x-large' title='" . attr_safe(_("Quiz page passed!")) . "'>&check;</span>";
>     $class = 'quiz-ok';
> }
> else
> {
>     $text = "<span class='x-large' title='" . attr_safe(_("Quiz page not passed!")) . "'>&#x2717;</span>";
>     $class = 'quiz-not-ok';
> }
> ```

Done.

I have not yet followed @chrismiceli's suggestion of using the actual characters for the '✗' and '✓', but I can do that easily enough if that's the way we want to go. If we do, we should probably also update it elsewhere in the code.

Reply:
> agree, if we go this way though, can we put unicode characters directly instead of their encodings? so '✗' and '✓'.

And now done.